### PR TITLE
specify aws role duration seconds for aws role

### PR DESCRIPTION
**Pull request by wojciech12**
## Description of the change

Introducing duraction_seconds for aws role.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (non-breaking change that adds documentation)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Pull Request is no longer marked as "draft"
- [ ] Reviewers have been assigned
- [ ] Changes have been reviewed by at least one other engineer
- [x] The target branch is `future` unless the change is going directly into production

### CODE REVIEW

File: spacelift/resource_aws_role.go [hunk 1]
```diff
@@ -218,7 +224,7 @@ func resourceAWSRoleSet(ctx context.Context, client *internal.Client, id string,
 	var mutation struct {
 		AttachAWSRole struct {
 			Activated bool `graphql:"activated"`
-		} `graphql:"stackIntegrationAwsCreate(id: $id, roleArn: $roleArn, generateCredentialsInWorker: $generateCredentialsInWorker, externalID: $externalID)"`
+		} `graphql:"stackIntegrationAwsCreate(id: $id, roleArn: $roleArn, generateCredentialsInWorker: $generateCredentialsInWorker, externalID: $externalID, durationSeconds: $duration_seconds)"`
```
Comment (adamconnelly) [inline spacelift/resource_aws_role.go:227]: nit: `$durationSeconds` instead of `$duration_seconds`

File: spacelift/resource_aws_role_test.go [hunk 2]
```diff
@@ -75,16 +76,18 @@ func TestAWSRoleResource(t *testing.T) {
...
 				}
 			`, randomID),
 				Check: Resource(
 					resourceName,
 					Attribute("id", IsNotEmpty()),
 					Attribute("module_id", Equals(fmt.Sprintf("test-module-%s", randomID))),
 					Attribute("generate_credentials_in_worker", Equals("false")),
-					AttributeNotPresent("external_id"),
+					Attribute("duration_seconds", Equals("942")),
+					Attribute("external_id", IsEmpty()),
```
Comment (adamconnelly) [inline spacelift/resource_aws_role_test.go:90]: Why did you need to change this from `AttributeNotPresent()` to `IsEmpty()`?

File: spacelift/resource_aws_role.go (hunk 1 above)
Comment (wojciech12) [reply spacelift/resource_aws_role.go]: +1

File: spacelift/resource_aws_role_test.go (hunk 2 above)
Comment (wojciech12) [reply spacelift/resource_aws_role_test.go:90]: @adamconnelly I read the resource immediately after the creation of spacelift_aws_role, before we did not check what the state of unset values is - https://github.com/spacelift-io/terraform-provider-spacelift/pull/302/files#diff-21ecb1e1c4fcd6d7a2f835a4b4158104cf0d96cbe887c9b8960f51184688d9a7R125 . 

In the mutation, we set externalID to "" when externalID is missing.

Comment (adamconnelly) [reply spacelift/resource_aws_role_test.go:90]: Cool - that makes sense.

**Review by adamconnelly — APPROVED** [review]
